lib/src/adapters/json_api.dart

```diff
@@ -55,8 +55,9 @@ class JsonApiAdapter extends Adapter with Http {
     }
     JsonApiManyDocument cached = peekMany(endpoint, ids);
     if (cached.length != ids.length) {
-      List<JsonApiDocument?> cachedDocs = cached.toList();
-      Iterable<String?> cachedIds = cachedDocs.map((doc) => doc!.id);
+      List<JsonApiDocument> cachedDocs = cached.toList();
+      Iterable<String> cachedIds =
+          cachedDocs.map((doc) => doc.id).whereType<String>().toList();
```

Comment:
@algodave whereType<String> is a nice alternative to cast, maybe we can make use of it in flutter-rest-data latest PR as well?

Reply:
@Bringoff Yes, definitely